test/library/standard/Endian/Endian.chpl

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2020 Hewlett Packard Enterprise Development LP
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ /*
+    Supports for little-endian and big-endian byte order conversions.
+
+    This module provides a Chapel interface wrapped over the standard
+    C endian module to support little-endian and big-endian byte order
+    conversions. It has functions to convert byte encoding of integer
+    values from the byte order of the host to and from little-endian
+    and big-endian formats.
+ */
+module Endian {
+  use SysCTypes;
+
+  private extern proc htole16(x: c_uint): c_uint;
+  private extern proc le16toh(x: c_uint): c_uint;
+  private extern proc htobe16(x: c_uint): c_uint;
+  private extern proc be16toh(x: c_uint): c_uint;
+  private extern proc htole32(x: c_uint): c_uint;
+  private extern proc le32toh(x: c_uint): c_uint;
+  private extern proc htobe32(x: c_uint): c_uint;
+  private extern proc be32toh(x: c_uint): c_uint;
+  private extern proc htole64(x: c_ulong): c_ulong;
+  private extern proc le64toh(x: c_ulong): c_ulong;
+  private extern proc htobe64(x: c_ulong): c_ulong;
+  private extern proc be64toh(x: c_ulong): c_ulong;
+
+  /*
+    Has value true if the host system is little-endian type.
+  */
+  const isLittleEndian = hostToLittleEndian(10) == 10;
+
+  /*
+    Has value true if the host system is big-endian type.
+  */
+  const isBigEndian = hostToBigEndian(10) == 10;
+
+  /* Returns the little endian order of the argument.
+
+      :arg x: a 16-bit signed integer or a 16-bit unsigned integer
+  */
+  proc hostToLittleEndian(x) where numBytes(x.type) == 2 {
```

Comment:
Users do what they want XD

Reply:
This will also resolve the `chpldoc` link problem because of same names I guess.

Reply:
That would also work (note that `private` already prevents symbols from being documented, so the "no doc" pragma would be redundant in that situation)

Reply:
> and throw a false or might an illegal argument error, if the argument does not match with our desired types.

@Aniket21mathur I'd try to avoid returning a runtime error (a return code or a thrown Error) to report a static type incompatability.  The user will only find out about it at runtime, and only if their execution happens to go down that path.

You can write a more comprehensive `where` clause than you have above,

```chapel
proc toHost(x: ?t) where (isSubtype(t, integral) || isSubtype(t, real)) && numBytes(t) is suitable {
```

(If I followed what types you want to accept, and written those subtypes correctly.)


Or if all the tests are of types and params, another option would be to call `compilerError()` in the unsupported cases.

```chapel
proc toHost(x: ?t) {
  if t is (int or real) and 8 bytes then
    handle 8 bytes
  else if t is (int or real) and 4 bytes then
    handle 4 bytes
  else if ...
    ...
  else
    compilerError("useful message");
}
```

which will cause a compilation error with a customized message if none of the earlier cases triggered.

I'm not an expert in these areas though.